### PR TITLE
Main Arm Board Complete

### PR DESCRIPTION
**Pull request by Brendanofawesome**
### The Central Arm Board is fully designed and routed
although the position of components is subject to change as the mechanical design of the arm is finalized. 

- I used exclusively the components we have in inventory (except for the TVS diode) and physically checked footprint dimensions.
- CAN and USB Data lines are length-matched. (I tried to make USB lines 90 Ohm, but I'm not really sure how that works tbh😭😭). 
- The board has M3 mounting holes at (5, 5), (25, 30), and (85, 55) mm and all components are on the front side for easy mounting. 

image
image

**Let me know if there is anything I should fix :)**

### CODE REVIEW

**Comment by DolphinGui** [comment]
First off: why are ground zones on top and bottom rather than in middle? The 40V plane is likely noisy, and will impact all traces above and below it. Typically the ground zone is placed in middle to block the noise. See https://www.allaboutcircuits.com/technical-articles/common-layer-stack-ups-for-a-four-layer-board/

**Comment by DolphinGui** [comment]
image
Secondly, it doesn't pass DRC. Please fix all of the DRC issues or explicitly explain why they should be ignored.

**Comment by DolphinGui** [comment]
Third, I didn't accurately describe how to connect CAN. There's actually no need for CAN to go back to the board every time. You can just daisy-chain the motors directly, which simplifies things.
can

**Comment by DolphinGui** [comment]
> Is it possible to have the schematic show up using some github action? That way its easier to review as a PR without needing to download the branch and bring up KiCad.

https://kicanvas.org/?github=https%3A%2F%2Fgithub.com%2FSJSURoboticsTeam%2Furc-electrical-2026%2Fblob%2FArm_Main%2FArm_Main%2FRobotics%2520Arm.kicad_pro
KiCanvas kind of works, but unfortunately doesn't show DRC or connections. I don't really think there's a good alternative to downloading the branch and viewing it directly.

**Comment by kammce** [comment]
That works pretty well to be honest. Especially for review. So a bot that could automatically provide links to view these would be great.

**Comment by kammce** [comment]
Also, I like the digital shunt on this design. Great idea!

**Comment by DolphinGui** [comment]
Also, I'd like your opinion on JST (crimped) connectors vs phoenix (terminal block) connectors. See [this document](https://docs.google.com/document/d/1k8CZk_FWZoye8f_co8tp2dPmifIb7WolZVLzof0EYnE/edit#heading=h.cfzpqjgaj1px). Literally everybody last meeting voted for terminal block, but I'd like to hear your opinion so we can build a consensus.

**Comment by kammce** [comment]
> Also, I'd like your opinion on JST (crimped) connectors vs phoenix (terminal block) connectors. See [this document](https://docs.google.com/document/d/1k8CZk_FWZoye8f_co8tp2dPmifIb7WolZVLzof0EYnE/edit#heading=h.cfzpqjgaj1px). Literally everybody last meeting voted for terminal block, but I'd like to hear your opinion so we can build a consensus.

My opinion or the @Brendanofawesome ? 

As for mine, so that it is known in a public space, I'm a big supporter of pluggable phoenix (terminal) blocks. They are very easy to crimp, very easy to determine if they are plugged in, easy to fix if there is a problem, support large currents if necessary. Overall, unless we really need to keep the size small Phoenix connectors are the way to go.

**Comment by Brendanofawesome** [comment]
Alright, I've (hopefully) fixed all the DRC errors and the CAN wiring. Here's how it all looks now:
image

**Comment by Brendanofawesome** [comment]
However, I might need some help with the 40v noise situation. Firstly, I left the 40V ground disconnected from the 5V ground. My reasoning is to avoid any situation where the 40V system passes power through the 5V connectors. I do know that the grounds are connected at some point, so should I just connect them here? Maybe with some sort of filter or resistor?

Secondly, as to board stack-up: Currently, I have a 40V+ and 40V- plane because I was worried about how wide the traces would need to be for the required current (>5mm). Should I just bite the bullet and them? Otherwise, how should I change the board stack-up? I have some ideas, but I'm not sure which is best or if there is a better option.
image

**Comment by DolphinGui** [comment]
> > Also, I'd like your opinion on JST (crimped) connectors vs phoenix (terminal block) connectors. See [this document](https://docs.google.com/document/d/1k8CZk_FWZoye8f_co8tp2dPmifIb7WolZVLzof0EYnE/edit#heading=h.cfzpqjgaj1px). Literally everybody last meeting voted for terminal block, but I'd like to hear your opinion so we can build a consensus.
> 
> My opinion or the @Brendanofawesome ?
> 
> As for mine, so that it is known in a public space, I'm a big supporter of pluggable phoenix (terminal) blocks. They are very easy to crimp, very easy to determine if they are plugged in, easy to fix if there is a problem, support large currents if necessary. Overall, unless we really need to keep the size small Phoenix connectors are the way to go.

@Brendanofawesome 's actually.

**Comment by DolphinGui** [comment]
Also, I probably should have mentioned this earlier, but I'm not convinced the 40V needs it's own plane. In the past we have routed the entire board on 2-layer boards with no noticeable noise issues from the 40V. In the past 2.5mm width traces have been used to route 40V, but I just use zones myself.

**Comment by DolphinGui** [comment]
As seen here:
image

**Comment by Brendanofawesome** [comment]
> Also, I'd like your opinion on JST (crimped) connectors vs phoenix (terminal block) connectors. See [this document](https://docs.google.com/document/d/1k8CZk_FWZoye8f_co8tp2dPmifIb7WolZVLzof0EYnE/edit#heading=h.cfzpqjgaj1px). Literally everybody last meeting voted for terminal block, but I'd like to hear your opinion so we can build a consensus.

I think moving away from JST in general is probably a good idea. They're kind of flimsy and I am worried about them coming loose during driving. If that means using Phoenix connectors instead then I agree :).

**Comment by Brendanofawesome** [comment]
I changed the motor connections into individual traces. Does the new design look good, or should I change the routing?

**Comment by DolphinGui** [comment]
You don't seem to have pushed any changes onto github. Did you forget to do that?

**Comment by Brendanofawesome** [comment]
OOP my badddd, I forgot to "push" lol.

Also would you like me to make a version of the board that uses Phoenix connectors instead?

**Comment by DolphinGui** [comment]
yes, switch the connectors on the board to phoenix connector.

**Comment by Brendanofawesome** [comment]
Here's what it looks like now :)
image

**Comment by DolphinGui** [comment]
I've somehow completely forgot to mention this, but is there a reason why BAT- is seperate from ground? I presume this is out of caution for noise, but not tying the grounds can result in a ground loop between the CAN bus and the motor. (This can be solved by doing galvanic isolation on the CAN, but that's way overkill and too complex for this ) Again, we've put the motor and the electronics on the same ground before, and our applications aren't that noise sensitive.

**Comment by Brendanofawesome** [comment]
> I've somehow completely forgot to mention this, but is there a reason why BAT- is seperate from ground? 

Ah, I was trying to ensure that current from the motors would never go through the small 5V line connector. I'm imagining that if the 5V- and 40V- connections are in parallel, then around 20% of the motor current will go through the 5V- line (assuming 18awg and 12awg wires respectively). I'm worried that the 5V connector will heat up.

I don't really understand how ground loops work in general, but if you think that will be a problem, then I can connect them :).

**Comment by Brendanofawesome** [comment]
> 2\. Why are there decoupling caps near the battery? [Batteries typically have near zero noise that would require that](https://electronics.stackexchange.com/questions/424364/what-is-the-typical-rms-noise-of-a-lithium-battery).

I was worried that the relatively long wires between the batteries and the board would cause problems because the motors draw large dI/dt

However, this is just a "it feels like they should be there" kind of thing rather than being based on any math..... I can remove them if they are unnecessary

**Comment by DolphinGui** [comment]
> > I've somehow completely forgot to mention this, but is there a reason why BAT- is seperate from ground?
> 
> Ah, I was trying to ensure that current from the motors would never go through the small 5V line connector. I'm imagining that if the 5V- and 40V- connections are in parallel, then around 20% of the motor current will go through the 5V- line (assuming 18awg and 12awg wires respectively). I'm worried that the 5V connector will heat up.
> 
> I don't really understand how ground loops work in general, but if you think that will be a problem, then I can connect them :).

The motors are not connected to 5V anyways, so no current will flow through 5V from the motor. Actually I'm not too sure what you mean by parallel, you'll have to draw out a diagram.



> > 2. Please switch the 90 degree connector in the bottom right corner to an XT60 that's in inventory.
> 
> I think we do have 90 degree xt60 connectors in inventory, but I can still switch it if you would like to keep all the connectors the same.

Are they? It's fine to keep them if they are, I just didn't see them in the inventory spreadsheet.



> > 2. Why are there decoupling caps near the battery? [Batteries typically have near zero noise that would require that](https://electronics.stackexchange.com/questions/424364/what-is-the-typical-rms-noise-of-a-lithium-battery).
> 
> I was worried that the relatively long wires between the batteries and the board would cause problems because the motors draw large di/dt
> 
> However, this is just a "it feels like they should be there" kind of thing rather than being based on any math..... I can remove them if they are unnecessary

I think what you're looking for is a bulk capacitor. Bulk capacitors are commonly used to provide power when the switching power supply is "switched" off. Usually for bulk capacitors aluminum electrolytic capacitors are used. That said I'm not convinced there's enough parasitic inductance to really make this an issue. Furthermore, the capacitors would have to be placed on the motors themselves to make this effective, since there would be more parasitic inductance in the wire between this and the motor.

**Comment by DolphinGui** [comment]
Also can you replace the TVS diode with the WE-TVS-824015043, then rename it to DT1446-04TS-7? The symbols are nearly the same and the footprint is *nearly* identical, just that the WE-TVS is in the built-in library.

**Comment by DolphinGui** [comment]
Also, cosmetic and hyper-nitpicky change:
Can the arm output connector symbols be replaced with non-mounting point ones so that the mounting points don't have to be connected?

**Comment by Brendanofawesome** [comment]
> The motors are not connected to 5V anyways, so no current will flow through 5V from the motor. Actually I'm not too sure what you mean by parallel, you'll have to draw out a diagram.

image

**Comment by Brendanofawesome** [comment]
> Are they? It's fine to keep them if they are, I just didn't see them in the inventory spreadsheet.

I'm pretty sure.... I might have to check again this Saturday
side note --- I've been trying to update the inventory spreadsheet but everything's kind of a mess🥲

**Comment by DolphinGui** [comment]
> > The motors are not connected to 5V anyways, so no current will flow through 5V from the motor. Actually I'm not too sure what you mean by parallel, you'll have to draw out a diagram.
> 
> image

Current flows inversely proportionally to resistance. Since 12 AWG wire has less resistance, it will bear more current than 18 AWG wire, even in parallel.

**Comment by Brendanofawesome** [comment]
> Current flows inversely proportionally to resistance. Since 12 AWG wire has less resistance, it will bear more current than 18 AWG wire, even in parallel.

Sounds goooddd. Doing the math it seems like the 18AWG wire will only carry about 20% of the current, so if the motors are drawing less than like 15A it should be fine :).

On to grounding --- I've heard that usually, you want to connect grounds together in more of a "star pattern". Should I do that here or should I just put them on the same net?
image

**Comment by Brendanofawesome** [comment]
Should I add a JTAG connector?

**Comment by DolphinGui** [comment]
Yes.

**Comment by DolphinGui** [comment]
> > Current flows inversely proportionally to resistance. Since 12 AWG wire has less resistance, it will bear more current than 18 AWG wire, even in parallel.
> 
> Sounds goooddd. Doing the math it seems like the 12AWG wire will only carry about 20% of the current, so if the motors are drawing less than like 15A it should be fine :).
> 
> On to grounding --- I've heard that usually, you want to connect grounds together in more of a "star pattern". Should I do that here or should I just put them on the same net? image

See https://resources.altium.com/p/pcb-ground-plane-best-practices-your-multilayer-stackup.
TLDR: In most cases, just have it be a uniform ground plane.

**Comment by DolphinGui** [comment]
Also see https://www.ti.com/lit/an/slyt499/slyt499.pdf and https://www.ti.com/lit/an/slyt512/slyt512.pdf

**Comment by DolphinGui** [comment]
Yes, I do get notifications every time you push. Also, I think you broke the CAN thing again. The arm board's supposed to be a terminating connection, and CAN is daisy chained through the wire instead of going through a hub.

**Comment by Brendanofawesome** [comment]
> The arm board's supposed to be a terminating connection, and CAN is daisy chained through the wire instead of going through a hub.

Wait sorry what do you mean by this? Does the CAN terminating resistor go on this board?

**Comment by DolphinGui** [comment]
> > The arm board's supposed to be a terminating connection, and CAN is daisy chained through the wire instead of going through a hub.
> 
> Wait sorry what do you mean by this? Does the CAN terminating resistor go on this board?

Yes. See 

> Third, I didn't accurately describe how to connect CAN. There's actually no need for CAN to go back to the board every time. You can just daisy-chain the motors directly, which simplifies things. can

**Comment by Brendanofawesome** [comment]
Ah, I intended for the second CAN connector to go to the stationary motor on the linear rail. Those wires will have to go in their own cable channel so I figured that it would be more organized to give it another connector instead of daisy chaining.
image

**Comment by DolphinGui** [comment]
> Ah, I intended for the second CAN connector to go to the stationary motor on the linear rail. Those wires will have to go in their own cable channel so I figured that it would be more organized to give it another connector instead of daisy chaining. image

My concern is that the wires will exceed 0.3m, which is the maximum stub length. See https://e2e.ti.com/support/interface-group/interface/f/interface-forum/378932/iso1050-can-bus-stub-length

**Comment by Brendanofawesome** [comment]
Hmmm I didn’t know that — uh oh. I’m thinking I might want to remove the end effector CAN connection and just daisy chain it off the motor connection. Do you think this would work? (The stub to the linear motor might have to be slightly longer than 0.3m)
IMG_2153

Other than that, thank you for all the suggestions and taking the time to look at everything :)

**Comment by DolphinGui** [comment]
> Hmmm I didn’t know that — uh oh. I’m thinking I might want to remove the end effector CAN connection and just daisy chain it off the motor connection. Do you think this would work? (The stub to the linear motor might have to be slightly longer than 0.3m) IMG_2153
> 
> Other than that, thank you for all the suggestions and taking the time to look at everything :)

I'm thinking that the linear motor is on the far end of the motor daisy chain. I'm also going to note that I'm not sure if the end effector will even have its own CAN connection, or be another I2C device again. Will need to check in with mechanical on how much is expected of the end effector.

**Comment by Brendanofawesome** [comment]
Hmmm, I think you're right in that maybe the end effector doesn't need CAN --- I didn't think about that. Although it does need I2C from what I understand :/.

Other than that, for the CAN connection to the motors, I just realized: wouldn't the motor connection be a giant stub in both cases because the termination resistor is on the PCB? Should I prepare to solder a TH resistor to the end of the daisy chain instead? 

also lwk realizing that CAN kinda sucks for modular systems, how feasible is something like 100M ethernet? (of course for next year, we're kind of too deep to switch to CAN now)

**Comment by DolphinGui** [comment]
Ok, I just realized that you were absent for all meetings where I explained this. 
can

All CAN bus connections are parallel to a single CAN bus that goes through the whole system. It begins from the drive system and snakes throughout the whole system until it ends at the arm/science board. Any motors, hub boards, or whatever that needs to communicate over CAN go in the middle. Alternatively if the end effector ends up needing CAN the bus ends at the end effector instead.

**Comment by DolphinGui** [comment]
Can you change the reset button to be a 2.54mm header so that it can be hooked up to external switches?

**Comment by DolphinGui** [comment]
I'm also going to note on the misapplication of via fencing here. Via fencing is used to absorb RF, either reducing noise from noisy high-speed digital lines or protecting sensitive analog lines. In this case they're kinda placed everywhere, even on lines that are not expected to generate RF or be sensitive. In any case more generally ground stitching is better than less, so it's probably better to keep it as is.

**Comment by Brendanofawesome** [comment]
> Can you change the reset button to be a 2.54mm header so that it can be hooked up to external switches?

Should I make that in parallel, or just replace the button altogether?

**Comment by Brendanofawesome** [comment]
> I'm also going to note on the misapplication of via fencing here. Via fencing is used to absorb RF, either reducing noise from noisy high-speed digital lines or protecting sensitive analog lines. In this case they're kinda placed everywhere, even on lines that are not expected to generate RF or be sensitive. In any case more generally ground stitching is better than less, so it's probably better to keep it as is.

Ah, I see what you mean. My intention was to place a via anywhere that a signal trace broke apart the ground pour. However, I was a bit sporadic with this and probably should have just used a via stitching plugin.

**Comment by DolphinGui** [comment]
> > Can you change the reset button to be a 2.54mm header so that it can be hooked up to external switches?
> 
> Should I make that in parallel, or just replace the button altogether?

either is fine

**Comment by DolphinGui** [comment]
> > I'm also going to note on the misapplication of via fencing here. Via fencing is used to absorb RF, either reducing noise from noisy high-speed digital lines or protecting sensitive analog lines. In this case they're kinda placed everywhere, even on lines that are not expected to generate RF or be sensitive. In any case more generally ground stitching is better than less, so it's probably better to keep it as is.
> 
> Ah, I see what you mean. My intention was to place a via anywhere that a signal trace broke apart the ground pour. However, I was a bit sporadic with this and probably should have just used a via stitching plugin.

that actually makes way more sense actually

**Comment by DolphinGui** [comment]
At this rate I think I'm going to order the arm board and the drive board. Both have been pretty done for a while, and it doesn't really look like any other boards are close to completion. Unless there are some last-minute changes you want to add, I'll tag this as revision 1 and order it on JLCPCB. From there it'll take about 2 weeks to arrive.

**Comment by Brendanofawesome** [comment]
That sounds good to me :)
I can’t think of anymore changes I would add

**Comment by Brendanofawesome** [comment]
Oh shoot wait we wired our connector pins opposite…. I’ll correct mine

**Comment by Brendanofawesome** [comment]
Should be all good now👍

**Review by DolphinGui — APPROVED** [review]
lgtm
